src/surfaceintersection.jl

```diff
@@ -0,0 +1,124 @@
+surfacecrossing(g_old::Taylor1{Taylor1{T}}, g::Taylor1{Taylor1{T}}, eventorder::Int) where {T <: Real} = g_old[eventorder+1][1]*g[eventorder+1][1] < zero(T)
+surfacecrossing(g_old::Taylor1{TaylorN{T}}, g::Taylor1{TaylorN{T}}, eventorder::Int) where {T <: Real} = g_old[eventorder+1][1][1]*g[eventorder+1][1][1] < zero(T)
+
+# An rudimentary convergence criterion for the Newton-Raphson root-finding process
+nrconvergencecriterion(g_val::T, nrabstol::T, nriter::Int, maxnriters::Int) where {T<:Real} = abs(g_val) > nrabstol && nriter ≤ maxnriters
+nrconvergencecriterion(g_val::Taylor1{T}, nrabstol::T, nriter::Int, maxnriters::Int) where {T<:Real} = abs(g_val[1]) > nrabstol && nriter ≤ maxnriters
+nrconvergencecriterion(g_val::TaylorN{T}, nrabstol::T, nriter::Int, maxnriters::Int) where {T<:Real} = abs(g_val[1][1]) > nrabstol && nriter ≤ maxnriters
+
+# This function should be moved to TaylorSeries.jl, but the definitive implementation still should be discussed
+function deriv(n::Int, a::Taylor1{T}) where {T <: Number}
+    @assert a.order ≥ n ≥ 0
+    if n==0
+        return a
+    elseif n==1
+        return derivative(a)
+    else
+        return deriv(n-1, derivative(a))
+    end
+end
+
+function taylorinteg(f!, g, q0::Array{U,1}, t0::T, tmax::T,
+        order::Int, abstol::T; maxsteps::Int=500, eventorder::Int=0,
+        maxnriters::Int=5, nrabstol::T=eps(T)) where {T <: Real,U <: Number}
+
+    # Allocation
+    const tv = Array{T}(maxsteps+1)
+    dof = length(q0)
+    const xv = Array{U}(dof, maxsteps+1)
+
+    # Initialize the vector of Taylor1 expansions
+    const t = Taylor1(T, order)
+    const x = Array{Taylor1{U}}(dof)
+    const dx = Array{Taylor1{U}}(dof)
+    const xaux = Array{Taylor1{U}}(dof)
+
+    # Initial conditions
+    @inbounds t[1] = t0
+    x0 = deepcopy(q0)
+    x .= Taylor1.(q0, order)
+    @inbounds tv[1] = t0
+    @inbounds xv[:,1] .= q0
+
+    #Some auxiliary arrays for root-finding/event detection/Poincaré surface of section evaluation
+    const g_val = zero(g(t,x,x))
+    const g_val_old = zero(g_val)
+    const slope = zero(U)
+    const dt_li = zero(U)
+    const dt_nr = zero(U)
+    const δt = zero(U)
+    const δt_old = zero(U)
+
+    const x_dx = vcat(x, dx)
+    const g_dg = vcat(g_val, g_val_old)
+    const x_dx_val = Array{U}( length(x_dx) )
+    const g_dg_val = vcat(evaluate(g_val), evaluate(g_val_old))
+
+    const tvS = Array{U}(maxsteps+1)
+    const xvS = similar(xv)
+    const gvS = similar(tvS)
+
+    #auxiliary Int for Newton-Raphson iterations
+    const nriter = 1
+
+    # Integration
+    const nsteps = 1
+    const nevents = 1 #number of detected events
+    const nextevord = eventorder+1
+    while t0 < tmax
+        δt_old = δt
+        δt = taylorstep!(f!, t, x, dx, xaux, t0, tmax, x0, order, abstol)
+        g_val = g(t, x, dx)
+        if surfacecrossing(g_val_old, g_val, eventorder)
+
+            #first guess: linear interpolation
```

Comment:
I guess you are looking for the value of the independent variable (t) such that `g` is zero. Why not use directly Newton-Raphson, starting at t=0? 

Also, I think that moving all the surface-crossing check to a separate function could make the code more readable.

Reply:
> I guess you are looking for the value of the independent variable (t) such that g is zero. Why not use directly Newton-Raphson, starting at t=0?

I was just looking for a "safe" starting point for the Newton-Raphson process, but yeah, that'd be another way to do it!

> Also, I think that moving all the surface-crossing check to a separate function could make the code more readable.

Nice suggestion, I'll do this right away!